File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/factory/HamburgesaFactory.java
package factory;

import models.Hamburgesa;
import models.Ingrediente;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class HamburgesaFactory {

    private HamburgesaFactory() {}
    private static HamburgesaFactory instance = null;

    public static HamburgesaFactory getInstance() {
        if(instance == null) instance = new HamburgesaFactory();
        return instance;
    }

    public Hamburgesa getRandom(List<Ingrediente> ingredientes){
        String[] nombres = {"Hambuergesa de vaca", "Hambuergesa de pollo", "Hambuergesa de cerdo"};
        List<Ingrediente> ingredientesHamburgesa = new ArrayList<>();
        int number = (int)(Math.random()*7+1);
        for(int i = 1; i<=number; i++){
            double precio = (Math.random()*100.0);
            ingredientesHamburgesa.add(
                    ingredientes.get((int) (Math.random() * ingredientes.size()))
            );
        }
        return new Hamburgesa(UUID.randomUUID(), nombres[(int)(Math.random()*nombres.length)], ingredientesHamburgesa);
    }
}

File: buerguer-pig/HamburgesasEIngredientes-Kotlin-IvanRC/src/main/kotlin/factory/HamburgesaFactory.kt
package factory

import model.Hamburgesa
import model.Ingrediente
import mu.KotlinLogging
import java.util.*

object HamburgesaFactory {

    private val logger = KotlinLogging.logger {}

    fun createRandom(ingredientes: List<Ingrediente>): Hamburgesa{
        val nombres = arrayOf("Hambuergesa de vaca", "Hambuergesa de pollo", "Hambuergesa de cerdo")
        val ingredientesHamburger = mutableListOf<Ingrediente>()
        val numero = (1..8).random()
        for(i in 1..numero){
            ingredientesHamburger.add(ingredientes.random())
        }
        return Hamburgesa(UUID.randomUUID(), nombres.random(), ingredientesHamburger)
    }
}